src/main/java/org/jenkinsci/plugins/workflow/support/actions/LogActionImpl.java

```diff
@@ -110,8 +101,8 @@
     private String charset;
 
     private LogActionImpl(FlowNode parent, Charset charset) {
-        if (!isRunning(parent)) {
-            throw new IllegalStateException("cannot start writing logs to a finished node " + parent);
+        if (!parent.isActive()) {
+            LOGGER.log(Level.WARNING, null, new IllegalStateException("cannot start writing logs to a finished node " + parent + " " + parent.getDisplayFunctionName() + " in " + parent.getExecution()));
```

Comment:
You're making an exception go away regardless, you'll forgive me if my wording isn't precise late at night.  In any case, the question I'd ask is: is there a legit case for instantiating when the parent isn't active?  If so, warn but no exception thrown.  

If not, then throw exception but use try/catch in the flow elsewhere so we don't break the build (positive impact: prevents other errors from breaking the build).